### DISCUSSION TITLE
🤖 معرفی اسکریپت نصب خودکار مرزبان من 🤖

### DISCUSSION THREAD

**Discussion by bitcoinvps** [Gozargah/Marzban#631] [General]
# معرفی اسکریپت نصب خودکار مرزبان من

می‌خواستم اسکریپت Bash‌ای که نوشته‌ام برای نصب و پیکربندی خودکار مرزبان را به اشتراک بگذارم. این اسکریپت کارهای زیر را انجام می‌دهد:

- نصب پیش‌نیازها مانند Docker، acme.sh و غیره.
- انتخاب بین نسخه توسعه دهندگان مرزبان یا آخرین نسخه.
- انتخاب بین پایگاه‌داده‌های sqlite و mysql.
- تولید و نصب گواهینامه‌های HTTPS با استفاده از acme.sh.  (بروزرسانی خودکار هر ۶۰ روز یکبار)
- نصب آسان PHPMyAdmin
- ساخت فایل docker-compose.yml.
- دانلود و پیکربندی آخرین نسخه XRAY-core.
- تولید شناسه‌های کوتاه و کلیدهای خصوصی تصادفی برای XRAY در هر بار نصب.
- تولید پیکربندی XRAY با انتخاب کاربر.
- نصب قالب‌های اشتراک سفارشی.
- پیکربندی پشتیبان‌گیری خودکار تلگرام.
- و موارد بیشتر!

## نحوه استفاده

برای استفاده از اسکریپت، فقط کافی است:

```
curl -sSL https://get.marzban.pro/ -o x.sh
chmod +x x.sh
./x.sh
```

اسکریپت سپس فرآیند نصب را شروع می‌کند و برای هر ورودی مورد نیاز از کاربر سؤال می‌کند.

## چرا این اسکریپت را ساختم

می‌خواستم راهی آسان برای راه‌اندازی سرورهای مرزبان با پیکربندی بهینه داشته باشم. اسکریپت قسمت‌های خسته‌کننده مانند صدور گواهینامه و به‌روزرسانی xray را به طور خودکار انجام می‌دهد. برای استقرارهای مرزبان من خیلی مفید بوده است!

اگر از این اسکریپت نصب استفاده کردید، خوشحال می‌شوم نظراتتان را بدانم. به هر گونه پیشنهاد برای بهبود کار علاقه‌مندم!

**Comment by iambabyninja** [comment]
Big thanx 4 your work ❤️

**Comment by abedin70** [comment]
ممنون دوست عزیز بابت زحمتی ک کشیدی.
آیا برای من که الان مرزبان نصب دارم هم کاربرد داره؟
یا باید از اول با این اسکریپت شما همرو نصب کنم؟
مثلا فقط برای دریافت سرتفکیت میشه از این استفاده کنم یا بقیه خدمات؟

**Comment by bitcoinvps** [comment]
> ممنون دوست عزیز بابت زحمتی ک کشیدی. آیا برای من که الان مرزبان نصب دارم هم کاربرد داره؟ یا باید از اول با این اسکریپت شما همرو نصب کنم؟ مثلا فقط برای دریافت سرتفکیت میشه از این استفاده کنم یا بقیه خدمات؟

در حال حاضر فقط برای نصب جدید کاربرد دارد.

**Comment by rezairp** [comment]
سلام دوست عزیز اسکریپت عالیه و مرسی که این اسکریپت رو درست کردی و کار رو خیلی راحت تر کردی
چندتا پیشنهاد دارم که اگه تو آپدیت بعدی اسکریپتت بزاری خیلی کاملتر میشه.یکی اینکه ساخت سایت جعلی روی دامنه رو اضافه کنی.دوم اینکه نصب و پیکربندی محدودیت تعداد کاربر رو بزاری و همچنین قابلیت تانل زدن رو اضافه کنی
به نظرم اگه این هارو به اسکریپت اضافه کنی خیلی کامل میشه
ممنون از اینکه به ما کمک میکنی

**Comment by bitcoinvps** [comment]
سلام و تشکر بابت بازخوردتون!

خوشحالم که از اسکریپت راضی هستید. در مورد پیشنهاداتتون، ممنون میشم اگر بتونید توضیحات بیشتری در مورد هرکدوم از درخواست‌ها بدین. برای مثال:

1. **ساخت سایت جعلی روی دامنه:** لطفاً توضیح دهید که دقیقاً چه نوع ویژگی‌ را در نظر دارید. آیا منابع یا آموزش‌های خاصی هست که بتوانند به ادراک من و توسعه این بخش کمک کنند؟

2. **محدودیت تعداد کاربر:** آیا منظور شما از محدودیت تعداد کاربر، ادغام `/mmdzov/luIP-marzban` با اسکریپت است؟ این می‌تواند جزئیات بیشتری نیاز داشته باشد تا به درستی پیاده‌سازی شود.

3. **قابلیت تانل زدن:** توجه داشته باشید که این ویژگی ممکن است نیاز به پیکربندی دو سرور داشته باشد و می‌تواند پیچیده باشد. آیا مراحل خاصی یا مدل‌های کاری مد نظرتان است که می‌تواند به توسعه این بخش کمک کند؟

با ارائه اطلاعات بیشتر، من می‌توانم بهتر درک کنم که چگونه این ویژگی‌ها را می‌توان به بهترین شکل در اسکریپت ادغام کرد. منتظر شنیدن نظرات و راهنمایی‌های شما هستم!

باز هم ممنون برای حمایتتون.

**Reply by Ali-Neo** [reply]
با نصب این اسکریپت نسخه جدید مرزبان هم نصب میشه ؟ یا هنوز نسخه ای قبلی رو نصب میکنه؟

**Reply by bitcoinvps** [reply]
همواره آخرین نسخه‌ مرزبان و آخرین نسخه XRAY را نصب می‌کند.

**Comment by hafez80** [comment]
میخوام ادمین غیر sudo اضافه کنم نمیشه
و نمیشه مرزبان رو ریستارت کرد
هیچکدوم از دستورات مرزبان روی این کار نمیکنه

**Reply by bitcoinvps** [reply]
در آپدیت بعدی سعی می کنم دستورات مرزبان را اضافه کنم. اگر به توضیحات بیشتر علاقه دارید، در اینجا توضیح داده ام: https://github.com/bitcoinvps/marzban-ezpz/issues/2#issuecomment-1820207903

**Reply by mmdmb1** [reply]
فایل تمپلیت را از کجا میشود ادیت کرد؟ کدوم مسیر فایلا ریخته میشه؟

**Reply by mmdmb1** [reply]
نحوه برگردوندن بکاپش چجوریه

**Reply by mmdmb1** [reply]
اینکه همرو میریزه تو پوشه روت از لحاظ امنیت خیلی جالب نیست راهی نداره درست بشه این داستانش؟

**Reply by bitcoinvps** [reply]
اسکریپت تمام فایل ها را در پوشه ای که اسکریپت را اجرا کرده اید ذخیره می کند. اگر آن را از پوشه root اجرا کرده اید، فایل ها را در پوشه root ذخیره می کند. 
در مورد بازیابی نسخه پشتیبان در واقع انجام آن آسان است. فایل فشرده را به سرور جدید منتقل کنید، docker را نصب کنید و سپس docker compose up -d را از پوشه ای که فایل فشرده را ذخیره و بازگشایی کردید اجرا کنید. با این کار بک آپ شما ظاهر می شود. 
به زودی اسکریپت بازیابی بکاپ را منتشر خواهم کرد.

**Reply by mmdmb1** [reply]
سلام بکاپ بر میگردونم یوزرا نمیاد بالا چه باید کرد؟

**Comment by mahtfa** [comment]
سلام موقع نصب بعد از طی شدن مراحل انتخاب inbound اررور زیر رو دریافت میکنم:
 ✔ Network ubuntu_mynetwork  Created                                                                                                                                                    0.1s
 ✔ Container mysql           Healthy                                                                                                                                                    0.1s
 ✘ Container marzban         Error                                                                                                                                                      0.0s
 ✔ Container nginx           Created                                                                                                                                                    0.0s
dependency failed to start: container marzban is unhealthy
تمامی پورت های سرور رو باز کردم تاثیری نداشت

**Reply by bitcoinvps** [reply]
سلام دوست من!
دستور زیر را اجرا کن و خروجی آن را به اشتراک بگذار:
`docker compose logs`

**Reply by mahtfa** [reply]
با دستور:
sudo docker compose logs > logs.txt
خروجی را توی فایل زیر ریختم
https://drive.google.com/file/d/1H9WEU-UCZQDf2DHKqhBSw-HJ29rRsatz/view?usp=sharing

**Comment by mohammadi939** [comment]
خیلی خوب بود و کارهای زیادی رو انجام میده.
دمت گرم

**Comment by mohammadi939** [comment]
آدرس فایل سِرت trojan ws tls مشکل داره، قسمت certificateFile آدرس اشتباه ثبت کردین و باید آدرس fullchain cert رو میدادین به جاش

`          "certificates": [
                        {
                            "ocspStapling": 3600,
                            "certificateFile": "/usr/local/etc/certs/example.com.cer",
                            "keyFile": "/usr/local/etc/certs/example.com.key"
                        }`

اونایی که مشکل دارن باید xray-config رو به این صورت تغییر بدن

`"certificateFile": "/usr/local/etc/certs/example.com.fullchain.cer",`

**Comment by mmdmb1** [comment]
سلام چطور میشه از دستورای مرزبان استفاده کرد تو این اسکریپت و یوزر ساخت برای پنل و..؟

**Comment by mahsamcse** [comment]
درود
ضمن تشکر از زحمت شما
دستور اول رو میزنم خطا میده
روی سرور ایران میزنم و شکن هم ست کردم
خطا:
curl: (35) error:0A000126:SSL routines::unexpected eof while reading
ممنون میشم راهنمایی بفرمایید

**Comment by iamtheted** [comment]
Hi 
**Do not install** Marzban with this script!
A lot of people have faced serious issues after installing Marzban with this.
The easy way is not always the best way to do something, just install Marzban with its official script.
